src/cascadia/TerminalSettingsModel/PowershellCoreProfileGenerator.cpp

```diff
@@ -309,7 +309,15 @@ void PowershellCoreProfileGenerator::GenerateProfiles(std::vector<winrt::com_ptr
     {
         const auto name = psI.Name();
         auto profile{ CreateDynamicProfile(name) };
-        profile->Commandline(winrt::hstring{ psI.executablePath.native() });
+
+        const auto& unquotedCommandline = psI.executablePath.native();
+        std::wstring quotedCommandline;
+        quotedCommandline.reserve(unquotedCommandline.size() + 2);
+        quotedCommandline.push_back(L'"');
+        quotedCommandline.append(unquotedCommandline);
+        quotedCommandline.push_back(L'"');
+        profile->Commandline(winrt::hstring{ quotedCommandline });

```

Comment:
for cheap tricks, we could also...

```
auto quotedCommandline = fmt::format(FMT_COMPILE(L"\"{}\""), psI.executablePath.native());
```

which _should_ (?????) compile down to effectively optimal code (!)

Reply:
A micro-benchmark on my machine shows my code to be faster :)

```cpp
        auto start = std::chrono::steady_clock::now();
        for (int i = 0; i < 100000; i++)
        {
            const auto& unquotedCommandline = psI.executablePath.native();
            std::wstring quotedCommandline;
            quotedCommandline.reserve(unquotedCommandline.size() + 2);
            quotedCommandline.push_back(L'"');
            quotedCommandline.append(unquotedCommandline);
            quotedCommandline.push_back(L'"');
        }
        auto end = std::chrono::steady_clock::now();
        std::wstringstream stream;
        stream << "Elapsed microseconds for native code " << std::chrono::duration_cast<std::chrono::microseconds>(end - start).count() << std::endl;
        OutputDebugString(stream.str().c_str());

        start = std::chrono::steady_clock::now();
        for (int i = 0; i < 100000; i++)
        {
            auto quotedCommandline = fmt::format(FMT_COMPILE(L"\"{}\""), psI.executablePath.native());
        }
        end = std::chrono::steady_clock::now();
        std::wstringstream stream2;
        stream2 << "Elapsed microseconds for fmt code " << std::chrono::duration_cast<std::chrono::microseconds>(end - start).count() << std::endl;
        OutputDebugString(stream2.str().c_str());
```

Output:
```
Elapsed microseconds for native code 11221
Elapsed microseconds for fmt code 15288
```

Reply:
> A micro-benchmark on my machine shows my code to be faster :)

Well, you'll make @lhecker 's day. And I chuckled. Good enough.

Reply:
LOL. Fair enough 😁